pkg/image/prune/prune.go

```diff
@@ -825,6 +829,10 @@ func (p *pruner) Prune(
 	errs := []error{}
 
 	errs = append(errs, pruneStreams(p.g, prunableImageNodes, streamPruner)...)
+	// if namespace is specified prune only ImageStreams and nothing more
```

Comment:
I'd avoid computing `graphWithoutPrunableImages` and `prunableComponents` if we prune only image streams.